### PR TITLE
 Fix indent comment alignment rule 3

### PR DESCRIPTION
**Pull request by KVic**
Also this fixes issue #1287.

### CODE REVIEW

File: src/indent.cpp [hunk 1]
```diff
@@ -2622,6 +2622,33 @@ static bool single_line_comment_indent_rule_applies(chunk_t *start)
...
+   chunk_t *next = pc;
+
+   do
+   {
+      chunk_t *next_nl = chunk_get_next(next);
+
+      if (next_nl == NULL || next_nl->nl_count > 1)
+      {
+         // FIXME: Max thresh magic number 5000
+         return(5000);
```
Comment (mihaipopescu) [inline src/indent.cpp]: I would use `int` and return -1 here to express an invalid value. Also, the name of the function could contain `abs` to clarify that.

There is no need to use `size_t` here.

**Comment by mihaipopescu** [comment]
Thanks for looking into this one.

File: src/indent.cpp (hunk 1 above)
Comment (KVic) [reply src/indent.cpp]: The int(-1) is also a magic number :). A std::optional should be used here, but this is a С++17. The magic number size_t(5000) is widely used in the options.cpp. It must be replaced everywhere with some constant, such as std::numeric_limits<size_t>::infinity().

Comment (mihaipopescu) [reply src/indent.cpp]: I agree with `-1`. I'm was just arguing against using `size_t`, and any other unsigned values, for anything other than sizes or bit-field operations. They introduce a lot of problems when used along with `int` due to implicit integer promotions.

**Pull request merged**